Cabal/doc/developing-packages.rst

```diff
@@ -2184,14 +2184,27 @@ system-dependent values for these fields.
        :pkg-field:`other-modules`, :pkg-field:`library:exposed-modules` or
        :pkg-field:`executable:main-is` fields.
 
+.. pkg-field:: hs-source-dir: directory list
+    :deprecated: 2.0
+    :removed: 3.0
+    :default: ``.``
+
+    Root directories for the module hierarchy.
+
+    Deprecated in favor of :pkg-field:`hs-source-dirs`.
+
 .. pkg-field:: hs-source-dirs: directory list
 
     :default: ``.``
 
     Root directories for the module hierarchy.
 
-    For backwards compatibility, the old variant ``hs-source-dir`` is
-    also recognized.
+    .. note::
+
+      Components can share source directories but modules will be compiled if
```

Comment:
"will be **re**compiled"?

Reply:
I did intend "compiled" here. Should I add "even if other components have already built these same modules?" Or would recompiled be clear/accurate enough?

Reply:
It just sounds kind of weird because what else would I expect cabal to do if not compile these modules ;)

I think the combination of both, "modules will be recompiled even if other components already built them" makes even more sense.

Reply:
Will rewrite this weekend.